lib/instrument/transformer/nskeyed.js

```diff
@@ -0,0 +1,389 @@
+  this.decodeArchive = function (archive) {
+    return archive.decode('NS.string');
+  };
+};
+
+UNARCHIVE_CLASS_MAP.NSMutableData = function () {
+  this.decodeArchive = function (archive) {
+    return archive.decode('NS.data');
+  };
+};
+
+UNARCHIVE_CLASS_MAP.NSUUID = function () {
+  this.decodeArchive = function (archive) {
+    const uuidBytes = archive.decode('NS.uuidbytes');
+    return uuidStringify(uuidBytes);
+  };
+  this.encodeArchive = function (obj, archive) {
+    archive._archive_obj['NS.uuidbytes'] = uuidParse(obj).buffer;
+  };
+};
+
+UNARCHIVE_CLASS_MAP.XCTCapabilities = function () {
+  this.decodeArchive = function (archive) {
+    return archive.decode('capabilities-dictionary');
+  };
+};
+
+UNARCHIVE_CLASS_MAP.DTTapMessageArchive = function () {
+  this.decodeArchive = function (archive) {
+    return archive.decode('DTTapMessagePlist');
+  };
+};
+
+UNARCHIVE_CLASS_MAP.ErrorArchive = function () {
+  this.decodeArchive = function (archive) {
+    return {'$class': 'NSError', 'domain': archive.decode('NSDomain'), 'userinfo': archive.decode('NSUserInfo'), 'code': archive.decode('NSCode')};
+  };
+};
+
+UNARCHIVE_CLASS_MAP.ExceptionArchive = function () {
+  this.decodeArchive = function (archive) {
+    return {'$class': 'NSException', 'reason': archive.decode('NS.reason'), 'userinfo': archive.decode('userinfo'), 'name': archive.decode('NS.name')};
+  };
+};
+
+UNARCHIVE_CLASS_MAP.XCTCapabilities = function () {
+  this.decodeArchive = function (archive) {
+    return archive.decode('capabilities-dictionary');
+  };
+};
+
+UNARCHIVE_CLASS_MAP.DTKTraceTapMessage = function () {
+  this.decodeArchive = function (archive) {
+    if (archive._object.$0) {
+      return archive.decode('$0');
+    } else {
+      return archive.decode('DTTapMessagePlist');
+    }
+  };
+};
+
+UNARCHIVE_CLASS_MAP.DTTapHeartbeatMessage = function () {
+  this.decodeArchive = DTTapMessagePlist;
+};
+UNARCHIVE_CLASS_MAP.DTSysmonTapMessage = function () {
+  this.decodeArchive = DTTapMessagePlist;
+};
+UNARCHIVE_CLASS_MAP.NSURL = NSURL;
+
+
+/**
+ * Capable of unpacking an archived object tree in the NSKeyedArchive format.
+ * Apple's implementation can be found here:
+ * https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSKeyedUnarchiver.swift
+ *
+ */
+class Unarchive {
+  constructor (inputBytes) {
+    this.input = inputBytes;
+    this.unpackedUids = {};
+    this.topUID = NULL_UID;
+    this.objects = [];
+  }
+
+  unpackArchiveHeader () {
+    const plist = plistlib.parseBuffer(this.input)[0];
+    if (plist.$archiver !== 'NSKeyedArchiver') {
+      log.error(`unsupported encoder: ${plist.$archiver}`);
+      return false;
+    }
+    if (plist.$version !== 100_000) {
+      log.error(`expected ${NSKeyedArchiveVersion}, got ${plist.$version }`);
+      return false;
+    }
+    const top = plist.$top;
+    const topUID = top.root;
+    if (!topUID) {
+      log.error(`top object did not have a UID! dump: ${top}`);
```

Comment:
if `top` is a plain object then something like `[Object object]` might be printed to the console. We might consider using `JSON.stringify()` to see what it actually contains